workflow/metrics/metrics_test.go

```diff
@@ -112,6 +112,10 @@ func TestMetricGC(t *testing.T) {
 	if assert.NoError(t, err) {
 		assert.Len(t, m.customMetrics, 1)
 	}
+	err = m.UpsertCustomMetric("realtime_metric", "workflow-uid", newCounter("test", "test", nil), true)
```

Comment:
I can see what you're doing with this test, but without this diff to read this test is slightly obscure. I know overlapping the two tests in time is good for speed of the run, but I'd still prefer to do the whole test isolated from the previous test for understandability.

Reply:
Ok, I add a new test: `TestRealtimeMetricGC`.